### DISCUSSION TITLE
Are there any tools like github copilot but for free (vscode extension)?

### DISCUSSION THREAD

**Discussion by ViniciusMarins** [community/community#122625] [Programming Help] [answered]
### Body

.

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by hafiz-muhammad** [comment]
Yes, one is called [Codeium](https://codeium.com/).

**Reply by ViniciusMarins** [reply]
Thanks!

**Comment by hafiz-muhammad** [comment]
> Thanks!

You're welcome.

**Comment by GM7Avila** [comment, marked as answer]
Bitu, Tabinine, and Amazon Codewhisperer are good tools. 

I used Amazon Codewhisperer for a while and had positive impressions.
